packages/warriorjs-core/src/Unit.js

```diff
@@ -92,10 +92,16 @@ class Unit {
    */
   performTurn() {
     if (this.isAlive()) {
-      this.effects.forEach(effect => effect.passTurn());
-      if (this.turn.action && !this.isBound()) {
-        const [name, args] = this.turn.action;
-        this.abilities.get(name).perform(...args);
+      this.effects.forEach(effect => { 
```

Comment:
This loop can be optimized and exit the `performTurn` method early if the `.isAlive()` check fails for one event.

Reply:
You could use `Array.some()`, but that'd lead to more obscure code. You can use `for..of` instead and just exit early with a `return` statement (`this.effects` is an instance of [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Iterating_Maps_with_for..of)).